src/ASM/ASMbase.h

```diff
@@ -922,6 +926,8 @@ class ASMbase
   //! in each parameter direction is set to \a p+nGauss%10.
   int nGauss; //!< \sa getNoGaussPt
 
+  ASM::CachePolicy cachePolicy = ASM::PRE_CACHE; //!< Basis function cache policy
```

Comment:
This is a global setting, I presume, so a copy of it on each patch seems a bit overloaded. Use a static variable instead?